internal/testing/test-summary/test-summary.go

```diff
@@ -76,27 +75,31 @@ func run(r io.Reader) (msg string, failures bool, err error) {
 		if err := json.Unmarshal(scanner.Bytes(), &event); err != nil {
 			return "", false, fmt.Errorf("%q: %v", scanner.Text(), err)
 		}
+
+		if event.Action == "fail" {
+			failedTests = append(failedTests, filepath.Join(event.Package, event.Test))
+		}
+
 		if *verbose && event.Action == "output" {
 			fmt.Print(event.Output)
 		}
-		// Ignore pass or fail events that don't have a Test; they refer to the
-		// package as a whole, and we would be over-counting if we included them.
-		// However, skips of an entire package are not duplicated with individual
-		// test skips.
-		if event.Test == "" && (event.Action == "pass" || event.Action == "fail") {
-			continue
-		}
-		counts[event.Action]++
-		if event.Action == "fail" {
-			failedTests = append(failedTests, filepath.Join(event.Package, event.Test))
+
+		// The Test field, if present, specifies the test, example, or benchmark
```

Comment:
Avoid phrases like "if present" and "not set". String fields are always present and set. Use "empty" and "non-empty".

Reply:
I copied it from `go doc test2json` :-)

But np, will change to "non-empty".